src/Engine/ProtoCore/FFI/CLRDLLModule.cs

```diff
@@ -10,6 +10,20 @@
 
 namespace ProtoFFI
 {
+
+    internal class TypeEqualityComparer : IEqualityComparer<Type>
+    {
+        public bool Equals(Type x, Type y)
+        {
+            return x.IsEquivalentTo(y) || x.Equals(y);
```

Comment:
dotnet provides this method to check if 2 embedded COM types are equivalent
https://learn.microsoft.com/en-us/dotnet/api/system.type.isequivalentto?view=net-7.0
IsEqual and hash code are different between 2 equivalent COM types